WMF Framework/Remote Notifications/Model/RemoteNotificationsModelController.swift

```diff
@@ -153,177 +141,162 @@ final class RemoteNotificationsModelController: NSObject {
         cache.currentUnreadCount = 0
         sharedCache.saveCache(cache)
     }
-
-    deinit {
-        NotificationCenter.default.removeObserver(self)
-    }
-
-    typealias ResultHandler = (Set<RemoteNotification>?) -> Void
     
-    public func newBackgroundContext() -> NSManagedObjectContext {
+    func newBackgroundContext() -> NSManagedObjectContext {
         let backgroundContext = persistentContainer.newBackgroundContext()
         backgroundContext.name = "RemoteNotificationsBackgroundContext"
         backgroundContext.automaticallyMergesChangesFromParent = true
         backgroundContext.mergePolicy = NSMergeByPropertyObjectTrumpMergePolicy
         return backgroundContext
     }
     
-    var numberOfUnreadNotifications: Int? {
+    //MARK: Count convenience helpers
+    
+    func numberOfUnreadNotifications() throws -> Int {
+        assert(Thread.isMainThread)
         let fetchRequest: NSFetchRequest<RemoteNotification> = RemoteNotification.fetchRequest()
         fetchRequest.predicate = unreadNotificationsPredicate
-        return try? viewContext.count(for: fetchRequest)
+        return try viewContext.count(for: fetchRequest)
+    }
+    
+    func numberOfAllNotifications() throws -> Int {
+        assert(Thread.isMainThread)
+        let fetchRequest: NSFetchRequest<RemoteNotification> = RemoteNotification.fetchRequest()
```

Comment:
I think here (and elsewhere within `RemoteNotificationsModelController` and `RemoteNotificationsRefreshOperation`) we can remove the `: NSFetchRequest<RemoteNotification>` and let type inference take care of us.